x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts

```diff
@@ -374,18 +374,41 @@ export class ManifestManager {
     newManifest: Manifest
   ): Promise<Error[]> {
     const errors: Error[] = [];
+
+    const artifactsToCreate: InternalArtifactCompleteSchema[] = [];
+
     for (const artifact of artifacts) {
       if (internalArtifactCompleteSchema.is(artifact)) {
-        const [err, fleetArtifact] = await this.pushArtifact(artifact);
-        if (err) {
-          errors.push(err);
-        } else if (fleetArtifact) {
-          newManifest.replaceArtifact(fleetArtifact);
-        }
+        artifactsToCreate.push(artifact);
       } else {
         errors.push(new EndpointError(`Incomplete artifact: ${getArtifactId(artifact)}`, artifact));
       }
     }
+
+    if (artifactsToCreate.length === 0) {
+      return errors;
+    }
+
+    let fleetArtifacts: InternalArtifactCompleteSchema[] = [];
+    try {
+      fleetArtifacts = await this.artifactClient.bulkCreateArtifacts(artifactsToCreate);
+    } catch (err) {
+      if (SavedObjectsErrorHelpers.isConflictError(err)) {
+        // TODO find artifact id from the bulk create error
```

Comment:
Something we should do before this lands or a follow-up?

Reply:
i'll correct this on the same pr, I found a way to reproduce the conflict